### PR TITLE
polish argo pkg

### PR DESCRIPTION
**Pull request by Yancey0623**
Polish the `pkg/argo`, design doc: https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/design/argo_workflow_on_sqlflow.md

code structure:
``` text
`-pkg/argo
  |-fetch.go        // Fetch API
  |-submit.go       // Submit API
  |-k8s_api.go      // Kuberntes API: kubectl create/logs/delete
  |-workflow.go     // Some helper functions to parse Argo workflow resource object.
```

TODO: #1459

### CODE REVIEW

File: pkg/proto/sqlflow.proto [hunk 1]
```diff
@@ -34,44 +34,39 @@ service SQLFlow {
...
-        RUNNING = 1;
-        SUCCEEDED = 2;
-        FAILED = 3;
-        ERROR = 4;
-        SKIPPED = 5;
-    }
-    FetchToken new_token = 1;
-    Phase phase = 2;
-    Logs logs = 3;
+    FetchRequest new_request = 1;
```
Comment (typhoonzero) [inline pkg/proto/sqlflow.proto]: why call it `new_request`

File: pkg/proto/sqlflow.proto [hunk 2]
```diff
@@ -34,44 +34,39 @@ service SQLFlow {
...
 }
 
-message FetchToken {
+message FetchRequest {
     Job job = 1;
     // the following fields keep the fetching state
     string step_id = 2;
     string log_offset = 3;
-    bool no_more_log = 4;  // True if no more logs
+    bool finish_fetching = 4;  // True if no more logs
```
Comment (typhoonzero) [inline pkg/proto/sqlflow.proto]: I prefer the word EOF here

Comment (typhoonzero) [reply pkg/proto/sqlflow.proto]: `finish_fetching` should be under `FetchResponse` but not in `Request`? We do not need to send `finish_fetching` to the server I think

Comment (Yancey0623) [reply pkg/proto/sqlflow.proto]: You're right, move `finish_fetching` to `FetchResponse`.

File: pkg/proto/sqlflow.proto (hunk 1 above)
Comment (Yancey0623) [reply pkg/proto/sqlflow.proto]: The client-side should keep the `FetchRequest` which contains the fetch status and refresh it after each `Fetch` gRPC request.

Comment (wangkuiyi) [reply pkg/proto/sqlflow.proto]: I see @typhoonzero 's point. How about we rename FetchRequest into FetchSince, and rename FetchResponse.new_request into FetchResponse.updated_fetch_since? @Yancey1989

File: pkg/proto/sqlflow.proto [hunk 3]
```diff
@@ -34,43 +34,38 @@ service SQLFlow {
...
-    // token = fetch_token(job=job)
-    // while True:
-    //     r = c.fetch(token)
-    //     print(r.logs.content)
-    //     if r.phase == SUCCEEDED/FAILED/ERROR and r.logs.no_more_log:
-    //         break
-    //     token = r.token
-    rpc Fetch (FetchToken) returns (FetchResponse);
+    // wfJob := Submit(argoYAML)
+    // req := newFetchRequest(wfJob)
```
Comment (wangkuiyi) [inline pkg/proto/sqlflow.proto]: From https://github.com/sql-machine-learning/sqlflow/pull/1488/files#diff-b03e31960a96575596cdfc09b491fefaR25, it seems that function `newFetchRequest` has 3 parameters.  However, in the pseudo-code here, it has only one.  How to set the rest two?

Comment (Yancey0623) [reply pkg/proto/sqlflow.proto]: The rest two would be the empty string, maybe the following code is clearer:

``` go
req := &FetchRequest {
  Job: &pb.Job{
    Id: wfJob,
   },
}
```

**Review by tonyyang-svail — APPROVED** [review]
LGTM.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**